cmd/nerdctl/network_create.go

```diff
@@ -96,34 +97,18 @@ func networkCreateAction(cmd *cobra.Command, args []string) error {
 	}
 	labels = strutil.DedupeStrSlice(labels)
 
-	if subnetStr == "" {
-		if gatewayStr != "" || ipRangeStr != "" {
-			return fmt.Errorf("cannot set gateway or ip-range without subnet, specify --subnet manually")
-		}
-	}
-
-	e, err := netutil.NewCNIEnv(globalOptions.CNIPath, globalOptions.CNINetConfPath)
-	if err != nil {
-		return err
-	}
-	createOpts := netutil.CreateOptions{
-		Name:        name,
-		Driver:      driver,
-		Options:     strutil.ConvertKVStringsToMap(opts),
-		IPAMDriver:  ipamDriver,
-		IPAMOptions: strutil.ConvertKVStringsToMap(ipamOpts),
-		Subnet:      subnetStr,
-		Gateway:     gatewayStr,
-		IPRange:     ipRangeStr,
-		Labels:      labels,
-	}
-	net, err := e.CreateNetwork(createOpts)
-	if err != nil {
-		if errdefs.IsAlreadyExists(err) {
-			return fmt.Errorf("network with name %s already exists", name)
-		}
-		return err
-	}
-	_, err = fmt.Fprintf(cmd.OutOrStdout(), "%s\n", *net.NerdctlID)
-	return err
+	return network.Create(types.NetworkCreateCommandOptions{
+		GOptions: globalOptions,
+		CreateOptions: netutil.CreateOptions{
+			Name:        name,
+			Driver:      driver,
+			Options:     strutil.ConvertKVStringsToMap(opts),
+			IPAMDriver:  ipamDriver,
+			IPAMOptions: strutil.ConvertKVStringsToMap(ipamOpts),
+			Subnet:      subnetStr,
+			Gateway:     gatewayStr,
+			IPRange:     ipRangeStr,
+			Labels:      labels,
+		},
+	}, cmd.OutOrStdout())
```

Comment:
nits: maybe we can keep the same code style with other command like https://github.com/containerd/nerdctl/blob/main/cmd/nerdctl/namespace_create.go#L39-L52(put the option construct process in a single function)

Reply:
I think it should be fine for now, since most the `commandAction` simply calls `processCommandOption` and `command.Command` (e.g., `network.Create`) in `pkg` side.

Later **if** we decide to move other non-core logic (those more related to cmd than core, e.g., option filtering/postprocess, formatting, outputting, etc) back to `cmd` side, we can put them in separate sub funcs.